module/VuFind/src/VuFind/View/Helper/Root/GoogleTagManager.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * GoogleTagManager view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\View\Helper\Root;
+
+/**
+ * GoogleTagManager view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class GoogleTagManager extends \Laminas\View\Helper\AbstractHelper
+{
+    /**
+     * Property ID (false if disabled)
+     *
+     * @var string|bool
+     */
+    protected $propertyId;
+
+    /**
+     * Constructor
+     *
+     * @param string|bool $propertyId Property ID (false if disabled)
+     */
+    public function __construct($propertyId)
+    {
+        $this->propertyId = $propertyId;
+    }
+
+    /**
+     * Returns GA code (if active) or empty string if not.
+     *
+     * @param string $customUrl override URL to send to Google Analytics
+     *
+     * @return string
+     */
+    public function __invoke($isHeadCode = false)
```

Comment:
Would it be more clear, rather than creating an __invoke method with a switch, to just have `getHeaderCode()` and `getFooterCode()` methods? If you leave the default __invoke from the parent class, its behavior is simply to return the helper, so then your template code would be `<?=$this->googleTagManager()->getHeaderCode()?>`, etc.

I don't insist on the change if you prefer this approach, but I feel that it might be more readable than a non-obvious boolean value.

Reply:
Your approach is much better.  I wasn't clear on the invoke() behavior.